File: caracal/sdk/enterprise/analytics.py
from caracal._version import get_version
"""
Copyright (C) 2026 Garudex Labs.  All Rights Reserved.
Caracal, a product of Garudex Labs

Analytics Extension (Enterprise Stub).

Advanced analytics export and dashboard integration.
In the open-source edition, all methods raise EnterpriseFeatureRequired.
"""

from __future__ import annotations

from typing import Any, Optional

from caracal.sdk.extensions import CaracalExtension
from caracal.sdk.hooks import HookRegistry
from caracal.sdk.enterprise.exceptions import EnterpriseFeatureRequired


class AnalyticsExtension(CaracalExtension):
    """Enterprise analytics export extension.

    Args:
        export_interval: Seconds between automatic metric exports.
    """

    def __init__(self, export_interval: int = 300) -> None:
        self._export_interval = export_interval

    @property
    def name(self) -> str:
        return "analytics"

    @property
    def version(self) -> str:
        return get_version()

    def install(self, hooks: HookRegistry) -> None:
        hooks.on_after_response(self._collect_metrics)

    def _collect_metrics(self, response: Any, scope: Any) -> None:
        raise EnterpriseFeatureRequired(
            feature="Analytics Metrics Collection",
            message="Advanced analytics requires Caracal Enterprise.",
        )

    def export(self, format: str = "json") -> Any:
        """Export analytics data."""
        raise EnterpriseFeatureRequired(
            feature="Analytics Export",
            message="Analytics data export requires Caracal Enterprise.",
        )

    def get_dashboard_url(self) -> str:
        """Get analytics dashboard URL."""
        raise EnterpriseFeatureRequired(
            feature="Analytics Dashboard",
            message="Analytics dashboard requires Caracal Enterprise.",
        )

File: sdk/node-sdk/src/enterprise/analytics.ts
/**
 * Copyright (C) 2026 Garudex Labs. All Rights Reserved.
 * Caracal, a product of Garudex Labs
 *
 * Analytics Extension (Enterprise Stub).
 * PROPRIETARY LICENSE — not covered by AGPLv3.
 */

import { CaracalExtension } from '../extensions';
import { HookRegistry } from '../hooks';
import { EnterpriseFeatureRequired } from './exceptions';

export class AnalyticsExtension implements CaracalExtension {
  readonly name = 'analytics';
  readonly version = '0.1.0';

  constructor(private readonly options?: { exportInterval?: number }) {}

  install(hooks: HookRegistry): void {
    hooks.onAfterResponse(() => {
      throw new EnterpriseFeatureRequired('Analytics Metrics Collection');
    });
  }

  export(_format?: string): unknown {
    throw new EnterpriseFeatureRequired('Analytics Export');
  }

  getDashboardUrl(): string {
    throw new EnterpriseFeatureRequired('Analytics Dashboard');
  }
}
